library/core/src/ptr/mod.rs

```diff
@@ -20,12 +23,8 @@
 //!   be *dereferenceable*: the memory range of the given size starting at the pointer must all be
 //!   within the bounds of a single allocated object. Note that in Rust,
 //!   every (stack-allocated) variable is considered a separate allocated object.
-//! * Even for operations of [size zero][zst], the pointer must not be pointing to deallocated
```

Comment:
FWIW this was meant to be a paragraph on ZST, not on deallocation, so the rewording kind of defeats its purpose.

Reply:
The reason I hoisted it out is that I can *only* rationalize this rule under strict provenance, where you can distinguish "reference to a ZST subfield" (can be invalidated by a deallocation) and "a magical forged allocation" (can't be invalidated by a deallocation).